api/services/GithubBuildStatusReporter.js

```diff
@@ -1,104 +1,106 @@
-const Github = require("github")
-const logger = require("winston")
-const url = require("url")
-const config = require("../../config")
-const { Build, Site, User } = require("../models")
+const Github = require('github');
+const logger = require('winston');
+const url = require('url');
+const config = require('../../config');
+
+const environment = require('../../services/environment');
+const { Build, Site, User } = require('../models');
+
+const loadSiteUserAccessToken = site => site.getUsers({
+  where: {
+    githubAccessToken: { $ne: null },
+    signedInAt: { $ne: null },
+  },
+  order: [['signedInAt', 'DESC']],
+  limit: 1,
+}).then((users) => {
+  const user = users[0];
+  if (user && user.githubAccessToken) {
+    return user.githubAccessToken;
+  }
+  throw new Error('Unable to find valid access token to report build status');
+});
+
+
+const loadBuildUserAccessToken = build => Build.findById(build.id, { include: [Site, User] })
+  .then((foundBuild) => {
+    const user = foundBuild.User;
+
+    if (user.githubAccessToken) {
+      return user.githubAccessToken;
+    }
+    return loadSiteUserAccessToken(foundBuild.Site);
+  });
+
+
+const authenticateGithubClient = (accessToken) => {
+  const client = new Github({ version: '3.0.0' });
+  client.authenticate({
+    type: 'oauth',
+    token: accessToken,
+  });
+  return client;
+};
+
+
+const sendCreateGithubStatusRequest = (githubClient, options) => new Promise((resolve, reject) => {
+  githubClient.statuses.create(options, (err, res) => {
+    if (err) {
+      reject(err);
+    } else {
+      resolve(res);
+    }
+  });
+});
+
 
 const reportBuildStatus = (build) => {
-  let site
+  let site;
 
   return new Promise((resolve, reject) => {
     if (!build || !build.commitSha) {
-      reject(new Error("Build or commit sha undefined. Unable to report build status"))
+      reject(new Error('Build or commit sha undefined. Unable to report build status'));
     } else {
-      resolve()
+      resolve();
     }
-  }).then(() => {
-    return Site.findById(build.site)
-  }).then(model => {
+  }).then(() => Site.findById(build.site)).then((model) => {
     if (!model) {
-      throw new Error("Unable to find a site for the given build")
+      throw new Error('Unable to find a site for the given build');
     }
-    site = model
-    return loadBuildUserAccessToken(build)
-  }).then(accessToken => {
-    githubClient = authenticateGithubClient(accessToken)
+    site = model;
+    return loadBuildUserAccessToken(build);
+  }).then((accessToken) => {
+    const githubClient = authenticateGithubClient(accessToken);
+
+    const context = environment.env === 'production'
```

Comment:
I am not sure that the `environment` service exposes that `env` var. That is used primarily to get values from the `federalist-${environment}-env` user provided service.

Instead, we may want to consider adding either an `app_env` config to `config/app.js`, or a `github_status_context` config to `config/build.js`.